src/material.rs

```diff
@@ -12,30 +12,49 @@ use {extensions, json, texture, Gltf};
 
 pub use json::material::AlphaMode;
 
-///  The material appearance of a primitive.
+lazy_static! {
+    static ref DEFAULT_MATERIAL: json::material::Material = Default::default();
+}
+
+/// The material appearance of a primitive.
 pub struct Material<'a> {
     /// The parent `Gltf` struct.
     gltf: &'a Gltf,
 
-    /// The corresponding JSON index.
-    index: usize,
+    /// The corresponding JSON index - `None` when the default material.
+    index: Option<usize>,
 
     /// The corresponding JSON struct.
     json: &'a json::material::Material,
 }
 
 impl<'a> Material<'a> {
     /// Constructs a `Material`.
-    pub fn new(gltf: &'a Gltf, index: usize, json: &'a json::material::Material) -> Self {
+    pub fn new(
+        gltf: &'a Gltf,
+        index: usize,
+        json: &'a json::material::Material,
+    ) -> Self {
         Self {
             gltf: gltf,
-            index: index,
+            index: Some(index),
             json: json,
         }
     }
 
-    /// Returns the internal JSON index.
-    pub fn index(&self) -> usize {
+    /// Constructs the default `Material`.
+    pub fn default(gltf: &'a Gltf) -> Self {
+        Self {
+            gltf: gltf,
+            index: None,
+            json: &DEFAULT_MATERIAL,
+        }
+    }
+
+    /// Returns the internal JSON index if this `Material` was explicity defined.
+    ///
+    /// This function returns `None` if the `Material` is the default material.
+    pub fn index(&self) -> Option<usize> {
```

Comment:
@bwasty Do you think this is a sensible strategy? I considered an enum representation at first but wasn't sure whether it would be actually beneficial:
```rust
enum Material<'a> {
    Default {
        gltf: &'a Gltf,
    },
    Custom {
        gltf: &'a Gltf,
        index: usize,
        json: &'a json::material::Material,
    },
}
```
```